processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -1365,7 +1374,10 @@ public Object apply(Rowboat input)
         metBuilder.setValueType(ValueType.COMPLEX);
         writeColumn(
             v9Smoosher,
-            new ComplexColumnPartSerde(metricColumn, complexType),
+            ComplexColumnPartSerde.legacySerializerBuilder()
+                                  .withTypeName(complexType)
+                                  .withDelegate(metricColumn)
+                                  .build(),
             metBuilder,
             metric
         );
```

Comment:
IndexMergerV9 make v9 index files directly, the main step are very like with IndexMerger except the v8 to v9 conversion step is no longer necessary.